plugins/movetoarchive.koplugin/main.lua

```diff
@@ -0,0 +1,159 @@
+                        return
+                    end
+                    self:openFileBrowser(last_copied_from_dir)
+                end,
+            },
+            {
+                text = _("Set archive directory"),
+                keep_menu_open = true,
+                callback =  self.setArchiveDirectory,
+            }
+        },
+    }
+end
+
+function MoveToArchive:moveToArchive()
+    local move_done_text = _("Book moved.\nDo you want to open it from the archive folder?")
+    self:commonProcess(true, move_done_text)
+end
+
+function MoveToArchive:copyToArchive()
+    local copy_done_text =_("Book copied.\nDo you want to open it from the archive folder?")
+    self:commonProcess(false, copy_done_text)
+end
+
+function MoveToArchive:commonProcess(is_move_process, moved_done_text)
+    if not archive_dir_path then
+        self:showNoArchiveConfirmBox()
+        return
+    end
+    local document_full_path = self.ui.document.file
+    self.ui:onClose()
+    local filename
+    last_copied_from_dir, filename = util.splitFilePathName(document_full_path)
+
+    logger.dbg("MoveToArchive: last_moved/copied_from_dir :", last_copied_from_dir)
+    move_to_archive_settings:saveSetting(last_copied_from_config_key, ("%s/"):format(last_copied_from_dir))
+
+    if is_move_process then
+        FileManager:moveFile(document_full_path, archive_dir_path)
+        FileManager:moveFile(DocSettings:getSidecarDir(document_full_path), archive_dir_path)
+    else
+        FileManager:copyFileFromTo(document_full_path, archive_dir_path)
+        FileManager:copyRecursive(DocSettings:getSidecarDir(document_full_path), archive_dir_path)
+    end
+    local dest_file = string.format("%s/%s", archive_dir_path, filename)
+    require("readhistory"):updateItemByPath(document_full_path, dest_file)
+    ReadCollection:updateItemByPath(document_full_path, dest_file)
+    -- Update last open file.
+    if G_reader_settings:readSetting("lastfile") == document_full_path then
+        G_reader_settings:saveSetting("lastfile", dest_file)
+    end
+
+    self:showConfirmBox(moved_done_text, _("Ok"), function () ReaderUI:showReader(archive_dir_path .. filename) end,
+        function () self:openFileBrowser(last_copied_from_dir) end)
+    ReadHistory:removeItemByPath(document_full_path)
+end
+
+
+function MoveToArchive:setArchiveDirectory()
+    require("ui/downloadmgr"):new{
+        onConfirm = function(path)
+            archive_dir_path = path
+            move_to_archive_settings:saveSetting(archive_dir_config_key, ("%s/"):format(path))
+            move_to_archive_settings:flush()
+        end,
+    }:chooseDir()
+end
+
+function MoveToArchive:showNoArchiveConfirmBox()
+    self:showConfirmBox(_("No archive directory.\nDo you want to set it now?"), _("Set archive folder"), self.setArchiveDirectory)
+end
+
+function MoveToArchive:showConfirmBox(text, ok_text, ok_callback, cancel_callback)
+    UIManager:show(ConfirmBox:new{
+        text = text,
```

Comment:
You might as well directly use `UIManager:show(ConfirmBox:new{})` anywhere you use MoveToArchive:showConfirmBox().
This function does not bring any added value to that.